### DISCUSSION TITLE
Re-encoding FLAC files with higher compression level

### DISCUSSION THREAD

**Discussion by bbaserdem** [beetbox/beets#4886] [Q&A]
I want to re-encode all flac files in my library. Their compression levels vary; but I want to reencode them in level 12 if they are not already. How would I go around doing this with beets? I am not sure if the encode plugin would be helpful with this.

**Comment by RollingStar** [comment]
There's convert_on_import or similar which I believe works through the convert plugin. I don't know if you can run convert on a whole library and have it replace the originals. There are plugins like beet alternatives which allows parallel duplicate libraries.

**Comment by apnea** [comment]
If your library is in one location `ffmpeg` would be an easy tool, with options `-acodec flac -compression_level  12`.  If that's no the case, you could use `beet` to create a copy of your library in one location and then use `ffmpeg` or similar.

But you should be aware that after level 4 you're likely to gain a few percent reduction in size at best, perhaps as little as a fraction of a percent depending on the source material. [https://sound.stackexchange.com/questions/41964/flac-compression-level-comparison-efficiency-analysis](url)
